Source/WebCore/platform/graphics/gstreamer/mse/MediaPlayerPrivateGStreamerMSE.cpp

```diff
@@ -123,6 +153,7 @@ MediaPlayerPrivateGStreamerMSE::MediaPlayerPrivateGStreamerMSE(MediaPlayer* play
     : MediaPlayerPrivateGStreamer(player)
 {
     GST_TRACE("creating the player (%p)", this);
+    videoDecoderLimitsDefaults(mediaMaxWidth, mediaMaxHeight, mediaMaxFrameRate);
```

Comment:
Call `videoDecoderLimitsDefaults()` only if parsing hasn't been attempted before. Log an error if parsing was unsuccessful. Set the "parsing has been attempted" flag to true inconditionally.

Reply:
this function has to be called inside MediaPlayerPrivateGStreamerMSE::supportsType because it is a static function which is called before creation of MediaPlayerPrivateGStreamerMSE

Reply:
Yes, but supportsType() can be called multiple times (to check multiple type/resolution variants). There's no need to parse the limit defaults every time. That's why I was suggesting the "parsing has been attempted" static flag, to parse the values specified at build time (ie: never changing at runtime) only once.

Reply:
Ah, I was not precisely in my comment, my intention was to signaling that I will move calling videoDecoderLimitsDefaults to supportsType with all your suggestions (which are perfectly fine as you described)